modules/views/components/civicrm.core.inc

```diff
@@ -2911,7 +2911,7 @@ function _civicrm_core_data(&$data, $enabled) {
 
   while ($dao->fetch()) {
     // call getTree using $dao->id as groupID, $dao->extends as entityType, with possible subtypes in $dao->extends_entity_column_value
-    $extendsContactSubtype = (('Contact' === $dao->extends) && !empty($dao->extends_entity_column_value));
+    $extendsContactSubtype = (in_array($dao->extends, array('Contact', 'Individual', 'Organization', 'Household')) && !empty($dao->extends_entity_column_value));
```

Comment:
You can use ```CRM_Contact_BAO_ContactType::basicTypePairs()``` instead, which will consider **user-defined** contact types and sub-types.